### ISSUE TITLE
Jerky map dragging behavior / Switch to SDL relative mouse mode

### ISSUE THREAD

**Issue by attet** [widelands/widelands#5410]
Dragging the map via right click has some peculiarities, at least on Windows.

First, the sensitivity is way too high compared to normal mouse movement. This seems to be caused by the mouse_compensate_warp_ mechanism described here: https://github.com/widelands/widelands/blob/master/src/wlapplication.cc#L1056
I'm sure it was added for a (still valid?) reason, but on windows it seemed to essentially double the sensitivity, due to the mouse event generated by SDL_WarpMouseInWindow having zero relative motion, thus needing no such compensation. I didn't notice any downsides by removing it, but like I said maybe it's needed on other platforms?

More problematically, the map tends to warp around and drift. Best illustrated by this short clip where I attempt to drag in a circular motion:

https://user-images.githubusercontent.com/1911416/167632270-74936823-96c8-40f6-8039-a37b77aa9b99.mp4

The clip is only 25fps which makes the issue markedly less noticeable, but you can still clearly see warping and unintentional moving away from where I started. After a wee bit of debugging I found relative mouse movement mode in SDL and gave it a try. Compare with this:

https://user-images.githubusercontent.com/1911416/167632329-e562f3f3-2076-4e7f-be43-85ca2473677c.mp4

Still, only 25fps video, but this is much better. Doesn't jump around, and stays roughly in the starting position (within the limits of my hand accuracy 😆).

Moreover, when run live, it gets better. Turns out in normal mode the mouse motion event frequency is only around 20ms, thus limiting us to 50fps. By switching to relative mode I got butter smooth dragging, making my 100Hz monitor the bottleneck; this is of course when zoomed in and/or there's not too much stuff to render, as the rendering is still quite resource intensive...  😆 

So here's a relatively small patch that actually ends up simplifying the logic by letting SDL do all the dragging magic. Now this is all tested only on Windows, so I would be interested to hear how this breaks on other platforms 😉. Also, are these issues present only on Windows to begin with?
https://github.com/attet/widelands/commit/2e024b05d96f81b5762037c7d2b64775f71cb401

**Comment by Noordfrees** [comment]
Since the system upgrade to Ubuntu 22.04 I'm also getting extremely fast mouse dragging. This did not happen on 21.10. 

With your patch, mouse dragging is slower than it used to be (maybe half as fast). I did not observe any regressions regarding mouse warping.

**Comment by attet** [comment]
> Since the system upgrade to Ubuntu 22.04 I'm also getting extremely fast mouse dragging. This did not happen on 21.10.
> 
> With your patch, mouse dragging is slower than it used to be (maybe half as fast). I did not observe any regressions regarding mouse warping.

Thanks. The relative speed change is not necessarily important, as the dragging behavior could've been 'incorrect' before.
I.e. does a mouse movement that results in, say, 10cm of cursor movement on map when **not dragging** result in the same distance of map travelled when holding down drag? Also, I suppose all sorts of 'mouse acceleration' features from the OS may or may not influence the results...

**Comment by Noordfrees** [comment]
I checked with xdotool (a tool to simulate user input events under X11), and the exact same mouse movement parameters result in the same distance whether or not the right mouse button is down. So the dragging distance is the same as the normal distance. However it nonetheless *feels* too slow given the considerable size of Widelands maps at normal zoom – now it takes you ages to get anywhere. It used to be the same with the arrow keys, and their speed was increased by a very large amount in consequence. So a 2× speedup over the patch would make it more useful IMHO.

Mouse pointer speed and acceleration OS settings are not relevant here, IMHO, since these affect the pointer speed in all situations equally even before the event is passed to SDL…

**Comment by tothxa** [comment]
Tested with my ancient system (Debian buster / oldstable with SDL 2.0.9). With master I can get it to jump with an external mouse if I move it really fast. With your patch it doesn't jump, but mouse acceleration is also disabled. Using the touchpad physical button, the slow travel from top to bottom of the pad makes the same map movement in both, but again, acceleration doesn't work with your patch. Couldn't make it jump with the touchpad though.

**Comment by Noordfrees** [comment]
IMHO we should fix this for v1.1 since the game is practically unplayable with the too-fast scrolling
